packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/EditCollaborator.spec.js

```diff
@@ -0,0 +1,309 @@
+          {
+            fileOwner: 'user0',
+            username: 'user1'
+          },
+          1
+        ),
+        modifiable: false,
+        firstColumn: false
+      })
+    })
+
+    it('should render collaborator edit options', () => {
+      const wrapper = getShallowMountedWrapper({ user: 'user0' })
+
+      const editOptions = wrapper.find(selectors.editOptions)
+      expect(editOptions.exists()).toBe(true)
+
+      expect(editOptions.props()).toMatchObject({
+        collaboratorsPermissions: {},
+        existingCollaboratorType: 'user',
+        existingRole: {
+          description: 'Download, preview and share',
+          label: 'Viewer',
+          name: 'viewer',
+          permissions: ['read', 'share']
+        },
+        expirationDate: null
+      })
+    })
+  })
+
+  describe('Action Buttons', () => {
+    it('should render the cancel button', () => {
+      const wrapper = getShallowMountedWrapper({ user: 'user0' })
+      const cancelButton = wrapper.find(selectors.cancelButton)
+      expect(cancelButton.exists()).toBe(true)
+      expect(cancelButton.text()).toBe('Cancel')
+    })
+
+    it('should render the save button', () => {
+      const wrapper = getShallowMountedWrapper({ user: 'user0' })
+      const saveButton = wrapper.find(selectors.saveButton)
+      expect(saveButton.exists()).toBe(true)
+      expect(saveButton.attributes().disabled).toBe('true')
+      expect(saveButton.text()).toBe('Save')
+    })
+
+    it('cancel button should work', async () => {
+      const cancelTrigger = jest.spyOn(Editcollaborator.methods, '$_ocCollaborators_cancelChanges')
+
+      const wrapper = getMountedWrapper({ user: 'user0' })
+      const cancelButton = wrapper.find(selectors.cancelButton)
+
+      expect(cancelTrigger).not.toHaveBeenCalled()
+      await cancelButton.trigger('click')
+      expect(cancelTrigger).toHaveBeenCalledTimes(1)
+    })
+
+    it('save button should work', async () => {
+      const saveTrigger = jest.spyOn(Editcollaborator.methods, '$_ocCollaborators_saveChanges')
+
+      const wrapper = getMountedWrapper({ user: 'user0' })
+      const saveButton = wrapper.find(selectors.saveButton)
+
+      const roleSelect = wrapper.findComponent(VueSelect)
+      expect(roleSelect.props('value')).toMatchObject(roles.viewer)
+      expect(wrapper.vm.selectedRole).toMatchObject(roles.viewer)
+
+      await roleSelect.vm.select(roles.editor)
+      expect(roleSelect.props('value')).toMatchObject(roles.editor)
+      expect(wrapper.vm.selectedRole).toMatchObject(roles.editor)
+
+      expect(saveTrigger).not.toHaveBeenCalled()
+      await saveButton.trigger('click')
+      expect(saveTrigger).toHaveBeenCalledTimes(1)
+    })
+
+    it('should not display saving in progress indicator by default', () => {
+      const wrapper = getMountedWrapper({ user: 'user0' })
+
+      const spinner = wrapper.find(selectors.savingInProgress)
+      expect(spinner.exists()).toBe(false)
```

Comment:
these all tests can use a same wrapper at top with spies predefined. no need to define separate wrapper for every test